### ISSUE TITLE
Suppress warning on missing excluded tag in `include` directive

### ISSUE THREAD

**Issue by rben01** [asciidoctor/asciidoctor#4230]
The `include` directive allows filtering for tags to include/exclude, as documented [here](https://docs.asciidoctor.org/asciidoc/latest/directives/include-tagged-regions/). Oftentimes I'll use a catch-all `// tag::exclude_me[]` tag to quickly and easily exclude content in included files, and then I'll just make all my `include` directives look like `include::file[tags=tags;to;keep;!exclude_me]`. Unfortunately, Asciidoctor warns when an excluded tag is not present in the included file, which means I can't really add `!exclude` to every single `include` directive in my main Asciidoctor file without getting lots of spurious warnings. Instead I have to only add it when the included file actually has `// tag::exclude_me[]` somewhere. In other words, to avoid this warning, I have to keep my `include`'s in sync with the contents of the included files -- if there is no `// tag::exclude_me[]` in the included file, then also shouldn't be an `;!exclude_me` in the `include`. (Alternatively I can put an empty `// tag::exclude_me[]` in every included file, but this seems suboptimal and may mean editing lots of existing Asciidoctor source files.)

Ideally, warnings would only be shown when *included* content was missing, not *excluded* content; missing excluded content seems like it should be logged at the INFO level. "The thing you didn't want to include didn't exist in the first place" is much less severe than "The thing you *wanted* to include couldn't be found". (This reminds me of the joke: "How would you like your coffee?" "Two sugars, no cream" "We're out of cream. Is  two sugars, no milk ok?")

**Comment by mojavelinux** [comment]
> warnings would only be shown when included content was missing, not excluded content; missing excluded content seems like it should be logged at the INFO level. 

This seems like a reasonable request. I'll need to have a look and see how feasible it is to implemented, but instinct tells me it shouldn't be difficult.

**Comment by mojavelinux** [comment]
It turns out this is a pretty easy change. It also doesn't break any existing tests, so we hadn't actually defined / solidified the existing behavior. And it's consistent with negating tags using a wildcard, which we never warn about if there are no tags to negate.

**Comment by mojavelinux** [comment]
I'm debating whether to backport this to 2.0.x or introduce it in 2.1.x. Thoughts?

**Comment by mojavelinux** [comment]
I've merged into main (for Asciidoctor 2.1). I'm still waiting on someone to make the case that this belongs in a patch release of Asciidoctor 2.0.

**Comment by rben01** [comment]
@mojavelinux Thanks Dan. I guess I'd like to have this in 2.0, but it also depends on when 2.1 is planned on being released.

**Comment by mojavelinux** [comment]
The decision has nothing to do with when 2.1 is released. It has to do with whether this change can be classified as a bug fix or whether it's an improvement that is considered breaking. Can we reasonably argue that the existing behavior is unexpected? Are we violating semantic versioning or not?

**Comment by rben01** [comment]
I'd argue that downgrading the warning to INFO would be a breaking change, which should take priority over the fact that (I think that) the current behavior of logging as WARNING is unexpected. Probably best not to back port it.

**Comment by mojavelinux** [comment]
Thanks for that insight.

To be clear, we are not downgrading the warning. We are removing it entirely.

**Comment by mojavelinux** [comment]
I did some more thinking about this, and here's my reasoning. The warning for an unused exclude tag was never documented and never tested. Therefore, we can say it was unspecified behavior. That gives us an opportunity to change it in a patch release. If people complain loudly about it, we can always roll it back in a subsequent patch release. In other words, let's go for it!

**Comment by mojavelinux** [comment]
Backport to v2.0.x complete.

**Comment by rben01** [comment]
@mojavelinux I'm secretly happy about this change, since I think it *should* be/have been present 2.0.x, but I'm worried about what it spells for Asciidoctor's compatibility guarantees. Is this project ok with making breaking changes to any and all unspecified/untested behaviors? I don't know how much of Asciidoctor's behavior is considered arbitrary and how much is considered part of the standard, but as a user I'd be annoyed if undocumented behavior I'd been relying on was pulled out from under me (not that I'd expect that anyone was relying on this particular behavior). If nothing else it'd probably be worth spelling out in the docs that only documented and/or tested behavior is considered part of the standard.

**Comment by mojavelinux** [comment]
> Is this project ok with making breaking changes to any and all unspecified/untested behaviors?

Yes. Unspecified and untested behavior is not official behavior.

> I'm worried about what it spells for Asciidoctor's compatibility guarantees.

If you don't trust me after developing this project for > 10 years, I don't know what else to tell you.

**Comment by rben01** [comment]
@mojavelinux Ok then, sounds good. Never meant to imply that I don't trust you 🙂

**Comment by mojavelinux** [comment]
For the record, I think you made a very convincing case that the existing behavior was not logical and that we could classify it not only as unspecified behavior, but also as unwanted behavior.
